src/core/thread/meshcop_dataset.cpp

```diff
@@ -239,6 +247,26 @@ ThreadError Dataset::Set(const otOperationalDataset &aDataset, bool aActive)
         Set(tlv);
     }
 
+    if (aDataset.mIsChannelMaskPage0Set)
+    {
+        MeshCoP::ChannelMaskTlv tlv;
+        MeshCoP::ChannelMaskEntry *entry;
+        uint8_t data[6];
+        tlv.Init();
+        tlv.SetLength(sizeof(MeshCoP::ChannelMaskEntry) + sizeof(aDataset.mChannelMaskPage0));
+
+        entry = reinterpret_cast<MeshCoP::ChannelMaskEntry *>(data);
+        entry->SetChannelPage(0);
```

Comment:
It is fine to assume a single Channel Mask Entry with Channel Page 0 since the OpenThread API only supports writing this channel mask type.